### PR TITLE
Sagemaker Hyperpod Training Adaptor For Nemo Release 1.2.0

### PR DESCRIPTION
**Pull request by rohithn1**
## Description
Adding support for PEFT finetuning with Deepseek R1

## Merge Checklist
Put an x in the boxes that apply. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request.

### General
 - [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
 - [x] I have run `pre-commit run --all-files` on my code. It will check for [this configuration](../.pre-commit-config.yaml).
 - [x] I have updated any necessary documentation, including [READMEs](../README.md) and API docs (if appropriate)
 - [x] I have verified the licenses used in the license-files artifact generated in the Python License Scan CI check. If the license workflow fails, kindly check the licenses used in the artifact.

### Tests
 - [x] I have run `pytest` on my code and all unit tests passed.
 - [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
 - [x] I have ran a training job with real data (huggingface dataloader instead of synthetic data) and validated that the loss value goes down.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by julianhr — APPROVED** [review]

File: src/hyperpod_nemo_adapter/patches/patch_llama_flash_attn_cp.py [hunk 1]
```diff
@@ -33,36 +34,49 @@
...
     te.attention.DotProductAttention.get_extra_state = patched_get_extra_state
-    # patch https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/models/llama/modeling_llama.py#L465
-    LlamaFlashAttention2.__init__ = patched_LFA2__init__
-    # patch https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/models/llama/modeling_llama.py#L473
-    LlamaFlashAttention2.forward = patched_LFA2_forward
     # patch https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/models/llama/modeling_llama.py#L918
     LlamaModel.forward = patched_LM_forward
     # patch https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/models/llama/modeling_llama.py#L198
     LlamaRotaryEmbedding.forward = patched_LRE_forward
+    if llama_flash_attn_patch:
```
Comment (tombousso) [inline src/hyperpod_nemo_adapter/patches/patch_llama_flash_attn_cp.py:75]: This seems like the wrong way to do this - we should just avoid calling apply_patch if we don't want to apply patch

**Review by jessech-en — APPROVED** [review]
